web/packages/shared/utils/assertUnreachable.ts

```diff
@@ -17,5 +17,11 @@
  */
 
 export function assertUnreachable(x: never): never {
-  throw new Error(`Unhandled case: ${x}`);
+  throw new UnhandledCaseError(x);
+}
+
+export class UnhandledCaseError extends Error {
+  constructor(public unhandledCase: any) {
```

Comment:
Maybe we could use `unknown` instead of `any`?

Reply:
Not significant, since we only print the field.
But if it is `any`, a thing like `unhandledCase.abc.def` is allowed - if `unknown` it isn't, so the type is more strict. 

I will leave it up to you.

Reply:
Ah, that makes total sense for a highly "generic" field such as `unhandledCase`. Changed `any` to `unknown`.